components/sso-hostobject/org.wso2.carbon.hostobjects.sso/src/main/java/org/wso2/carbon/hostobjects/sso/internal/builder/AuthReqBuilder.java

```diff
@@ -106,4 +185,150 @@ private static NameIDPolicy buildNameIDPolicy() {
         nameIDPolicy.setAllowCreate(true);
         return nameIDPolicy;
     }
+
+    /**
+     * Sign the SAML AuthnRequest message
+     *
+     * @param authnRequest
+     * @param signatureAlgorithm
+     * @param cred
+     * @return
+     */
+    public static AuthnRequest setSignature(AuthnRequest authnRequest, String signatureAlgorithm,
+            X509Credential cred) throws Exception {
+        try {
+            Signature signature = (Signature) buildXMLObject(Signature.DEFAULT_ELEMENT_NAME);
+            signature.setSigningCredential(cred);
+            signature.setSignatureAlgorithm(signatureAlgorithm);
+            signature.setCanonicalizationAlgorithm(SignatureConstants.ALGO_ID_C14N_EXCL_OMIT_COMMENTS);
+            try {
+                KeyInfo keyInfo = (KeyInfo) buildXMLObject(KeyInfo.DEFAULT_ELEMENT_NAME);
+                X509Data data = (X509Data) buildXMLObject(X509Data.DEFAULT_ELEMENT_NAME);
+                org.opensaml.xml.signature.X509Certificate cert =
+                        (org.opensaml.xml.signature.X509Certificate) buildXMLObject(
+                                org.opensaml.xml.signature.X509Certificate.DEFAULT_ELEMENT_NAME);
+                String value = Base64.encodeBytes(cred.getEntityCertificate().getEncoded());
+                cert.setValue(value);
+                data.getX509Certificates().add(cert);
+                keyInfo.getX509Datas().add(data);
+                signature.setKeyInfo(keyInfo);
+            } catch (CertificateEncodingException e) {
+                throw new Exception("Error getting certificate", e);
```

Comment:
Better to use simple class name whenever possible, because it improves readability. (This is regarding the use of org.opensaml.xml.signature.X509Certificate instead of X509Certificate).